spark-extension/src/main/java/org/apache/spark/sql/auron/AuronConf.java

```diff
@@ -52,6 +52,9 @@ public enum AuronConf {
     // TypedImperativeAggregate one row mem use size
     UDAF_FALLBACK_ESTIM_ROW_SIZE("spark.auron.udafFallback.typedImperativeEstimatedRowSize", 256),
 
+    /// enable trimming string inputs before casting to numeric types
+    CAST_STRING_TRIM_ENABLE("spark.auron.cast.stringTrimBeforeNumeric", true),
```

Comment:
i suggest a better name `spark.auron.cast.trimString` because casting string to boolean also needs to be supported.

Reply:
Updated to `spark.auron.cast.trimString`